### ISSUE TITLE
Can't build with latest libressl

### ISSUE THREAD

**Issue by dmitryvakulenko** [mongodb/mongo-php-driver#920]
I get this errors when I try to install mongodb from the pecl on Alpine linux 3.8 

    /tmp/pear/temp/mongodb/src/libmongoc/src/libmongoc/src/mongoc/mongoc-stream-tls-openssl-bio.c:58:1: error: static declaration of 'BIO_set_data' follows non-static declaration
    ...
    /tmp/pear/temp/mongodb/src/libmongoc/src/libmongoc/src/mongoc/mongoc-stream-tls-openssl-bio.c:64:1: error: static declaration of 'BIO_get_data' follows non-static declaration
    ...
    /tmp/pear/temp/mongodb/src/libmongoc/src/libmongoc/src/mongoc/mongoc-stream-tls-openssl-bio.c:70:1: error: static declaration of 'BIO_set_init' follows non-static declaration

Everything build fine before package libressl-dev in Alpne was version 2.6.

Thank you.

**Comment by rufinus** [comment]
Confirmed - same here.

**Comment by derickr** [comment]
Hi!

It looks like this happens in our bundled version of [libmongoc](https://github.com/mongodb/mongo-c-driver), which has already seen a [fix](https://github.com/mongodb/mongo-c-driver/pull/526/files) for this. I don't know quite in which versions of libmongoc this has been committed too, as I can't find their ticket. I'm trying to figure out whether this also made it into their stable branch (1.12), which we currently use for the stable version (1.5.x) of the PHP driver.

As a temporary workaround, applying the patch from https://github.com/mongodb/mongo-c-driver/pull/526/files to ``src/libmongoc/src/libmongoc/src/mongoc/mongoc-stream-tls-openssl-bio.c`` should work, providing that you have the source code checked out with for example ``pecl download mongodb && tar xvzf mongodb-1.5.2.tgz && cd mongodb-1.5.2``.

cheers,
Derick

**Comment by dmitryvakulenko** [comment]
Thanks @derickr! Patch helped to me.

**Comment by derickr** [comment]
There is now a ticket for the CDRIVER bug: https://jira.mongodb.org/browse/CDRIVER-2808 — but we're still trying to find out how we get that bug fix into the PHP driver.

**Comment by rufinus** [comment]
after applying the patch i get

```
/mongodb-1.5.2/src/libmongoc/src/libmongoc/src/mongoc/mongoc-stream-tls-openssl.c:50:1: error: static declaration of 'BIO_meth_free' follows non-static declaration
 BIO_meth_free (BIO_METHOD *meth)
 ^~~~~~~~~~~~~
In file included from /mongodb-1.5.2/src/libmongoc/src/libmongoc/src/mongoc/mongoc-stream-tls-openssl.c:25:0:
/usr/include/openssl/bio.h:339:6: note: previous declaration of 'BIO_meth_free' was here
 void BIO_meth_free(BIO_METHOD *biom);
      ^~~~~~~~~~~~~
make: *** [Makefile:489: src/libmongoc/src/libmongoc/src/mongoc/mongoc-stream-tls-openssl.lo] Error 1
```

any hints?

Part from my Dockerfile:
```
 && pecl download mongodb \
 && tar xfvz mongodb-1.5.2.tgz \
 && cd mongodb-1.5.2 \
 && patch src/libmongoc/src/libmongoc/src/mongoc/mongoc-stream-tls-openssl-bio.c </mongodb.patch \
 && phpize \
 && ./configure && make && make install \

```

**Comment by derickr** [comment]
@rufinus — there are three files that needs patching, not just ``mongoc-stream-tls-openssl-bio.c``: https://github.com/mongodb/mongo-c-driver/pull/526/files

**Comment by rufinus** [comment]
thanks @derickr

**Comment by jmikola** [comment]
This should be addressed by the [1.5.3 release](https://github.com/mongodb/mongo-php-driver/releases/tag/1.5.3), which includes a bump to libmongoc 1.13.0.
